tests/unit/test_parameters/test_parameter_sets_class.py

```diff
@@ -25,7 +25,7 @@ def test_all_registered(self):
         """Check that all parameter sets have been registered with the
         ``pybamm_parameter_sets`` entry point"""
         known_entry_points = set(
-            ep.name for ep in pkg_resources.iter_entry_points("pybamm_parameter_sets")
+            ep.name for ep in il.entry_points()["pybamm_parameter_sets"]
```

Comment:
Looks like `importlib.metadata` has had some API changes in the past: this part gives a deprecation warning on newer Python versions (3.10 and above). Here are the logs:

1. https://github.com/pybamm-team/PyBaMM/actions/runs/6224348861/job/16919897538?pr=3335#step:11:1335
2. https://github.com/pybamm-team/PyBaMM/actions/runs/6224348861/job/16919897976?pr=3335#step:11:1333

could you change it and add a conditional? Something like

```python
if sys.version_info < (3, 10)
    # something
else:
```

should do it

Reply:
Ok I will take a look at fixing it

Reply:
Also, importing `importlib.metadata` as `il` is not very conventional, you could do `from importlib.metadata import ...`

Reply:
I imported importlib as "il" due to the generic naming of "entry_points". I figured an object with that name could easily be created in a project this size. I tend to be cautious about potential name clashes or confusing names.

I can change that as well when I push the fix above.